Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionMapsView.cs

```diff
@@ -57,43 +62,49 @@ public ActionMapsView(VisualElement root, StateContainer stateContainer)
             m_ListView.RegisterCallback<ValidateCommandEvent>(OnValidateCommand);
 
             CreateSelector(s => new ViewStateCollection<string>(Selectors.GetActionMapNames(s)),
-                (actionMapNames, state) => new ViewState(Selectors.GetSelectedActionMap(state), actionMapNames));
+                (actionMapNames, state) => new ViewState(
+                    Selectors.GetSelectedActionMap(state), actionMapNames?.ToList() ?? new List<string>()));
 
             m_AddActionMapButton = root.Q<Button>("add-new-action-map-button");
             m_AddActionMapButton.clicked += AddActionMap;
             ContextMenu.GetContextMenuForActionMapListView(this, m_ListView.parent);
         }
 
-        public override void RedrawUI(ViewState viewState)
+        protected override void RedrawUI(ViewState viewState)
         {
-            m_ListView.itemsSource = viewState.actionMapNames?.ToList() ?? new List<string>();
-            if (viewState.selectedActionMap.HasValue)
+            m_ListView.itemsSource = viewState.actionMapNames;
+            m_ListView.Rebuild();
+
+            // Update view to reflect model selection
+            var desiredSelectedIndex = viewState.selectedActionMapIndex;
+            if (desiredSelectedIndex >= 0 && desiredSelectedIndex != m_ListView.selectedIndex)
```

Comment:
Ah, I believe I understand what you are after now. Redraw seems to be scheduled via UI scheduler so I would expect user initiated and callback events being processed through the same scheduler queue but I might be wrong. I did not notice any such behavior while testing this but I know too little about UITK internals to be sure.

Reply:
A redras is triggered automatically as soon as the serializedObject changed as well, I think the scheduler could collide with this behavior, that's my doubt. If so this would mean we discard the selection of the list/treeView at a point. But I don't know certainly as well how this would be working, maybe worth to investigate with devs-uitoolkit.